go/signedexchange/verifier.go

```diff
@@ -353,18 +352,8 @@ func verifyTimestamps(sig *Signature, verificationTime time.Time) error {
 }
 
 func verifyPayload(e *Exchange, signature *Signature) ([]byte, error) {
-	var integrityStr string
-	var enc mice.Encoding
-	switch e.Version {
-	case version.Version1b1:
-		enc = mice.Draft02Encoding
-		integrityStr = "mi-draft2"
-	case version.Version1b2, version.Version1b3:
-		enc = mice.Draft03Encoding
-		integrityStr = "digest/" + enc.ContentEncoding()
```

Comment:
I was wondering if this change keeps this behavior.

Reply:
Any counter examples? `IntegrityIdentifier()` returns `"digest/mi-sha256-03"` for `Draft03Encoding`, which matches `"digest/" + enc.ContentEncoding()`.

Reply:
Hm, I could not see the assumption from this PR.

Reply:
Well, the original code assumed that the digest algorithm and the HTTP content encoding share the same name. This holds for [http-mice-03](https://tools.ietf.org/html/draft-thomson-http-mice-03#section-6), but it's not necessarily true.